webapp/src/components/sidebar/sidebarSettingsMenu.tsx

```diff
@@ -49,6 +52,11 @@ const SidebarSettingsMenu = React.memo((props: Props) => {
         setRandomIcons(!randomIcons)
     }
 
+    const [preferredDateFormat, setPreferredDateFormat] = useState(UserSettings.preferredDateFormat)
+    const updatePreferredDateFormat = (newDateFormat: string) => {
+        setPreferredDateFormat(newDateFormat)
```

Comment:
I think you also need to set the setting in the local storage here. something like `UserSettings.preferredDateFormat = newDateFormat`.

Reply:
Oh yes. thanks. Silly me thought that state dispatch will call call the setter!